lib/workers/repository/util/gfm.ts

```diff
@@ -0,0 +1,46 @@
+const UNICODE_WJ = '\u2060';
+const DASH_DASH = '--';
+const DASH_WJ_DASH = `-${UNICODE_WJ}-`;
+const RE_DASH_DASH = new RegExp(DASH_DASH, 'g');
+const RE_DASH_WJ_DASH = new RegExp(DASH_WJ_DASH, 'g');
+
+export const escapeGfmCommentText = (text: string): string => {
+  if (text.indexOf(DASH_WJ_DASH) !== -1) {
+    // Not even counting rare use of complex Unicode symbols overall, the exact combination of '-' <Unicode WJ> '-' is not making much sense
+    throw new Error(
+      "Text already contains '-' <Unicode WJ> '-' sequence or is already escaped"
+    ); // ... so it's fair to expect to never encounter it in any real text and this error to only be encountered on already escaped text
+  }
+
+  if (text.indexOf(DASH_DASH) === -1) {
+    // fast path
+    return text;
+  }
+
+  let previousText;
+  let resultText = text;
+
+  do {
+    previousText = resultText;
+    resultText = previousText.replace(RE_DASH_DASH, DASH_WJ_DASH); // '--' not allowed inside github flavored markdown comments
+  } while (resultText !== previousText);
+
+  return resultText;
+};
+
+export const unescapeGfmCommentText = (text: string): string => {
+  if (text.indexOf(UNICODE_WJ) === -1) {
+    // fast path
+    return text;
+  }
+
+  let previousText;
+  let resultText = text;
+
+  do {
+    previousText = resultText;
+    resultText = previousText.replace(RE_DASH_WJ_DASH, DASH_DASH);
+  } while (resultText !== previousText);
```

Comment:
No loop required when using `global` regex flag. see above

Reply:
> I don't really like this solution.
> 
> We should replace all `--` when we compute and slugify the branchname. that wouhld be a more robust and better solution

But then
1) The change will be a breaking change or will require a lot of additional code to intelligently recognize migration from an older version(we don't leave any mark of the version that produced the text of dashboard) to handle preexisting text.
2) No matter what escape character we use the text from the original issue text becomes really noisy.